### PR TITLE
Bst

### PR DESCRIPTION
**Pull request by SammyRobensParadise**
@maryrsmcphee check out what I did here i think it worked

### CODE REVIEW

File: Lab3/binary_search_tree.cpp [hunk 1]
```diff
@@ -228,10 +228,11 @@ bool BinarySearchTree::remove(BinarySearchTree::TaskItem val) {
 // TODO: Only works when first called
 bool BinarySearchTree::remove(BinarySearchTree::TaskItem *node, int k) {
     // if leaf node
-    if ((node->left == NULL && node->right == NULL) && (k == node->priority)) {
+    TaskItem *target = node;
+    if ((node->left == NULL && node->right == NULL) && (k == target->priority)) {
         cerr << "leaf node \n" << endl;
```
Comment (SammyRobensParadise) [inline Lab3/binary_search_tree.cpp]: here

File: Lab3/binary_search_tree.cpp [hunk 2]
```diff
@@ -297,10 +298,7 @@ bool BinarySearchTree::remove(BinarySearchTree::TaskItem *node, int k) {
         } else if (node->right->left == NULL && node->right->right != NULL) {
             // the node has one right child;
             TaskItem *temp = node->right->right;
-            node->right->right = node;
-            node = temp;
-            node->right->right = NULL;
-            free(node->right->right);
+            node->right = node->right->right;
```
Comment (SammyRobensParadise) [inline Lab3/binary_search_tree.cpp]: here

File: Lab3/tests.h [hunk 3]
```diff
@@ -222,7 +222,9 @@ class BinarySearchTreeTest {
         ASSERT_TRUE(bst.get_size() == 4);
         cerr<<" - Print: test4"<<endl;
         bst.print();
-        //  remove nodes
+        ASSERT_TRUE(bst.remove(t4));
```
Comment (SammyRobensParadise) [inline Lab3/tests.h:226]: this passes now

**Comment by maryrsmcphee** [comment]
i'll pull it

**Comment by SammyRobensParadise** [comment]
@maryrsmcphee  look at the latest commit as well

File: Lab3/tests.h (hunk 3 above)
Comment (maryrsmcphee) [reply Lab3/tests.h:226]: that's good

**Comment by maryrsmcphee** [comment]
test 8 still fails?

**Comment by SammyRobensParadise** [comment]
=

> test 8 still fails?

yea it likely will, this was simply for removing a node with one child, specifically the right child

**Comment by SammyRobensParadise** [comment]
@maryrsmcphee do you want me to merge?

**Comment by SammyRobensParadise** [comment]
@maryrsmcphee take a look at test7, lemme know if you think we are missing anything, but it lgtm

**Comment by maryrsmcphee** [comment]
looks great!

**Review by maryrsmcphee — APPROVED** [review]

**Pull request merged**